site/content/contribute/server/_index.md

```diff
@@ -20,10 +20,31 @@ The server consists of several different Go packages:
 
 * `api4` - Version 4 of the RESTful JSON Web Service
 * `app` - Logic layer for getting, modifying, and interacting with models
+* `services` - Set of packages providing functionality to the rest of the application, especially to the `app` package.
 * `cmd` - Command line interface
 * `einterfaces` - Interfaces for Enterprise Edition features
 * `jobs` - Job server and scheduling
 * `model` - Definitions and helper functions for data models
 * `store` - Storage layer for interacting with caches and databases
 * `utils` - Utility functions for various tasks
 * `web` - Serves static pages
+
+## New packages rules
+
+We are trying to follow certain rules about when is needed to create a new
+package or add functionality to the existing ones (Some of them can not fit
+with the current implementation because it is an ongoing effort).
+
+* Any integration with a third party tool should be implemented as an
+  independent service (Examples: Bleve, Cache, File backends).
+* Any well isolated chunk of code that has a very small public API and a lot of
```

Comment:
You are pointing out a very good practice here: "Very Small Public API". Actually this should be applicable for any package. This would definitely help our effort on isolating tests and speed up the test process. I think we can emphasize this as a standalone rule here.

Reply:
Do you propose to add an extra rule saying something like "Any package should expose the minimal API needed, trying to avoid any public function or method that is not used from the outside"?

Reply:
Exactly :)

Reply:
Great, I'll add that, I think makes a lot of sense here.